### ISSUE TITLE
Smoldyn simulation failing

### ISSUE THREAD

**Issue by bilalshaikh42** [reproducible-biomedical-modeling/CRBM-Viz#2641]
Failed sim:  https://run.biosimulations.dev/simulations/60ca36c5ec4fbe336094932b

Log: 
```
INFO:    Using cached SIF image
(EE)
Fatal server error:
(EE) Server is already active for display 1
        If this server is no longer running, remove /tmp/.X1-lock
        and start again.
(EE)
/usr/local/lib/python3.9/site-packages/biosimulators_utils/warnings.py:31: BioSimulatorsWarning: The SED document is potentially incorrect.
  - Model `model` may be invalid.
    - The model file `model.txt` may be invalid.
      - No validation is available for models encoded in `urn:sedml:language:smoldyn`
  warnings.warn(termcolor.colored(message, Colors.warning.value), category)
/usr/local/lib/python3.9/site-packages/biosimulators_utils/warnings.py:31: BioSimulatorsWarning: The COMBINE/OMEX archive may be invalid.
  - The SED-ML file at location `./simulation.sedml` may be invalid.
    - Model `model` may be invalid.
      - The model file `model.txt` may be invalid.
        - No validation is available for models encoded in `urn:sedml:language:smoldyn`
  warnings.warn(termcolor.colored(message, Colors.warning.value), category)
 succeeded

  Executed 1 tasks and 2 outputs:
    Tasks:
      Succeeded: 1
      Skipped: 0
      Failed: 0
    Outputs:
      Succeeded: 2
      Skipped: 0
      Failed: 0

Bundling outputs ...
Cleaning up ...

============= SUMMARY =============
Executed 1 SED documents:
  SED documents (1):
    Succeeded: 1
    Skipped: 0
    Failed: 0
  Tasks (1):
    Succeeded: 1
    Skipped: 0
    Failed: 0
  Outputs (2):
    Succeeded: 2
    Skipped: 0
    Failed: 0

/xvfb-startup.sh: line 6: kill: (3413906) - No such process
srun: error: xanadu-55: task 0: Exited with exit code 1
srun: Terminating job step 3065144.1
```

**Comment by bilalshaikh42** [comment]
Potential solution here: https://github.com/ssandrews/Smoldyn/pull/91

**Comment by jonrkarr** [comment]
The same solution should be copied to 
- [x] `Biosimulations/tools/xvfb-startup.sh` fixed in #2632 
- [x] `Biosimulators_pyNeuroML/scripts/xvfb-startup.sh`

**Comment by jonrkarr** [comment]
The Smoldyn image should be fixed by biosimulators/Biosimulators#255

**Comment by jonrkarr** [comment]
@bilalshaikh42 could you try the Smoldyn run again with the revised image?

**Comment by bilalshaikh42** [comment]
Updated error

```
=============Running docker image for simulator=============
INFO:    Using cached SIF image
(EE) 
Fatal server error:
(EE) Server is already active for display 1
	If this server is no longer running, remove /tmp/.X1-lock
	and start again.
(EE) 
/usr/local/lib/python3.9/site-packages/biosimulators_utils/warnings.py:31: BioSimulatorsWarning: The SED document is potentially incorrect.
  - Model `model` may be invalid.
    - The model file `model.txt` may be invalid.
      - No validation is available for models encoded in `urn:sedml:language:smoldyn`
  warnings.warn(termcolor.colored(message, Colors.warning.value), category)
/usr/local/lib/python3.9/site-packages/biosimulators_utils/warnings.py:31: BioSimulatorsWarning: The COMBINE/OMEX archive may be invalid.
  - The SED-ML file at location `./simulation.sedml` may be invalid.
    - Model `model` may be invalid.
      - The model file `model.txt` may be invalid.
        - No validation is available for models encoded in `urn:sedml:language:smoldyn`
  warnings.warn(termcolor.colored(message, Colors.warning.value), category)
 succeeded

  Executed 1 tasks and 2 outputs:
    Tasks:
      Succeeded: 1
      Skipped: 0
      Failed: 0
    Outputs:
      Succeeded: 2
      Skipped: 0
      Failed: 0

Bundling outputs ...
Cleaning up ...
```

**Comment by jonrkarr** [comment]
I see the message, but the simulation runs and the final error appears to be gone. Does the run actually fail?

To cleanup the error message, we could try adding the following before `Xvfb $DISPLAY`

```sh
if [ -f "/tmp/.X1-lock" ]; then
  rm /tmp/.X1-lock
fi
```

**Comment by bilalshaikh42** [comment]
Yes, the run step is marked as failed on slurm

**Comment by jonrkarr** [comment]
I think this can be closed now. 

I mistakenly pushed the same image again. I fixed this. I was able to run a simulation successfully (https://run.biosimulations.org/simulations/60cca9fe69720db446b98c68).

Regarding the first error (`If this server is no longer running ...`), I haven't been able to replicate this locally. I don't see this message in the log for `60cca9fe69720db446b98c68` either. If we see this again, I think we could try running with a different value of the `DISPLAY` environment variable: `singularity run --env "DISPLAY=:1000" ...`? If that helps, we can reflect that change in the Docker image.
